### PR TITLE
Move to separate compilation units

### PR DESCRIPTION
**Pull request by akva2**
Some small low-hanging fruits that can be moved to separate compilation units to avoid recompilation in each simulator object.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/flow/countGlobalCells.hpp [hunk 1]
```diff
@@ -21,28 +21,26 @@
   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#ifndef OPM_BLACKOILDETAILS_HEADER_INCLUDED
-#define OPM_BLACKOILDETAILS_HEADER_INCLUDED
+#ifndef OPM_COUNTGLOBALCELLS_HEADER_INCLUDED
+#define OPM_COUNTGLOBALCELLS_HEADER_INCLUDED
+
+#include <config.h>
```
Comment (bska) [inline opm/simulators/flow/countGlobalCells.hpp:27]: It is always an error to include `<config.h>` in headers.

File: opm/simulators/flow/countGlobalCells.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    std::iota(all_cells.begin(), all_cells.end(), 0);
+    return all_cells;
+}
+
+double getGravity(const double* g, const int dim) {
+    double grav = 0.0;
+    if (g) {
+        // Guard against gravity in anything but last dimension.
+        for (int dd = 0; dd < dim - 1; ++dd) {
+            assert(g[dd] == 0.0);
```
Comment (bska) [inline opm/simulators/flow/countGlobalCells.cpp:40]: You should include `<cassert>` to get a declaration of the `assert` macro in scope.

**Review by bska — COMMENTED** [review]
Looks good.  Just couple of minor nits and then this is ready for merging.  Thanks a lot for taking on this task.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks for the updates.  I'll merge this once the build check is green.

**Comment by bska** [comment]
> I'll merge this once the build check is green.

Build check is green now.  Merging as advertised.

**Pull request merged**